src/tss2-fapi/ifapi_json_eventlog_serialize.c

```diff
@@ -795,10 +791,24 @@ bool ifapi_json_TCG_SPECID_EVENT_cb(
     callback_data *cb_data = data;
     json_object *jso_event_list = cb_data->jso_event_list;
     TCG_SPECID_EVENT *event_specid = (TCG_SPECID_EVENT*)tcg_event->event;
+    bool use_pcr0 = false;
+    size_t i;
 
     if (cb_data->skip_event)
         return true;
 
+    if (cb_data->pcr_list_size) {
+        for (i = 0; i < cb_data->pcr_list_size; i++) {
+            if (cb_data->pcr_list[i] == 0)
+                use_pcr0 = true;
+        }
+    } else {
+        use_pcr0 = true;
+    }
```

Comment:
This block of code is repeated, should we make it a function?

Reply:
yes that would be better. I will add a function.